packages/block-library/src/navigation/style.scss

```diff
@@ -611,18 +611,16 @@ button.wp-block-navigation-item__content {
 		}
 	}
 
-	@include break-small() {
-		&:not(.hidden-by-default) {
-			&:not(.is-menu-open) {
-				display: block;
-				width: 100%;
-				position: relative;
-				z-index: auto;
-				background-color: inherit;
-
-				.wp-block-navigation__responsive-container-close {
-					display: none;
-				}
+	:not(.is-collapsed) > & {
+		&:not(.is-menu-open) {
```

Comment:
I feel this warrants a comment. It's confusing to read and analyse which makes it a perfect candidate.

Reply:
Done in ad7533893c